test/api_test.js

```diff
@@ -576,6 +633,84 @@ describe('Api', () => {
     });
   });
 
+  describe('credentials', () => {
+    async function testErrorReturn(fn, doc) {
+      try {
+        await fn.apply(client, [{signature: pkcs7TestData.signedData, document: doc}]);
+      } catch (err) {
+        return err.code;
+      }
+
+      throw new Error(`Function ${fn} should have failed`);
+    }
+
+    let createTemporaryCredentials = taskcluster.createTemporaryCredentials;
+
+    let expectedCredentials = {
+      clientId: 'myClientId',
+      accessToken: 'myAccessToken',
+      certificate: 'myCertificate',
+    };
+
+    let doc = JSON.parse(Buffer.from(pkcs7TestData.doc, 'base64').toString());
+
+    before(() => {
+      taskcluster.createTemporaryCredentials = sandbox.stub();
```

Comment:
For the sinon sandboxing to work (`sandbox.restore()`) this should to be done slightly differently.  In other places, we create stubs() directly because we're not stubbing APIs, instead using the stub directly.  For the sandboxing to restore the taskcluster module correctly, sinon itself must do the assignment so that it can track it to revert it on `.restore()`.  This code might fix it, but I haven't tested it.

```javascript
   let stub = sandbox.stub(taskcluster, 'createTemporaryCredentials');
   stub.returns(expectedCredentials);
```